plugins/logitech-tap/fu-logitech-tap-hdmi-device.c

```diff
@@ -0,0 +1,507 @@
+		FuChunk *chk = g_ptr_array_index(chunks, i);
+		g_autoptr(GByteArray) data_pkt = g_byte_array_new();
+		/* device expect fixed size buffer, so cannot leverage actual size here: fu_chunk_get_data_sz(chk) */ 
+		g_byte_array_append(data_pkt, fu_chunk_get_data(chk), kLogiDefaultImageBlockSize);
+		if (!fu_logitech_tap_hdmi_device_set_xu_control(self,
+			kLogiUnitIdVidCapExtension,
+			kLogiUvcXuAitCustomCsSetFwData,
+			data_pkt->len,
+			(guchar *)data_pkt->data,
+			error))
+			return FALSE;
+		fu_progress_step_done(progress);
+	}
+
+	/* uninit */ 
+	if (!fu_logitech_tap_hdmi_device_ait_finalize_update(self,
+			error))
+		return FALSE;
+
+	/* signal for sensor device to trigger composite device reboot */
+	fu_device_add_private_flag(device, FU_LOGITECH_TAP_HDMI_DEVICE_FLAG_NEEDS_REBOOT);
+	return TRUE;
+}
+
+static gboolean
+fu_logitech_tap_hdmi_device_set_version(FuDevice *device, GError **error)
+{
+	FuLogitechTapHdmiDevice *self = FU_LOGITECH_TAP_HDMI_DEVICE(device);
+	guint16 data_len = 0;
+	g_autofree guint8 *query_data = NULL;
+    g_autofree gchar *fwversion_str = NULL;
+    guint8 set_data[XU_INPUT_DATA_LEN] = {kLogiTapHdmiVerSetData,0,0,0,0,0,0,0};
+
+	g_debug("get HDMI firmware version");
+
+	if (!fu_logitech_tap_hdmi_device_set_xu_control(self,
+						      kLogiUnitIdVidCapExtension,
+						      kLogiTapCameraVersionSelector,
+						      XU_INPUT_DATA_LEN,
+						      (guchar *)set_data,
+						      error))
+		return FALSE;
+
+	data_len = 0;
+	/* query current device version */
+	if (!fu_logitech_tap_hdmi_device_query_data_size(self,
+						       kLogiUnitIdVidCapExtension,
+						       kLogiHdmiVerGetSelector,
+						       &data_len,
+						       error))
+		return FALSE;
+	if (data_len > XU_INPUT_DATA_LEN) {
+		g_set_error(error,
+			    G_IO_ERROR,
+			    G_IO_ERROR_FAILED,
+			    "version query packet was too large at 0x%x bytes: ",
+			    data_len);
+		return FALSE;
+	}
+
+	query_data = g_malloc0(data_len);
+	if (!fu_logitech_tap_hdmi_device_get_xu_control(self,
+						      kLogiUnitIdVidCapExtension,
+						      kLogiHdmiVerGetSelector,
+						      data_len,
+						      (guchar *)query_data,
+						      error))
+		return FALSE;
+
+	/*  little-endian data. MajorVersion byte 3&2, MinorVersion byte 5&4, BuildVersion byte 7&6 */
+	fwversion_str = g_strdup_printf("%i.%i.%i", query_data[3] + query_data[2] * 100, query_data[5] + query_data[4] * 100, query_data[7] + query_data[6] * 100);
+	fu_device_set_version(device, fwversion_str);
+
+	/* success */
+	return TRUE;
+}
+
+static gboolean
+fu_logitech_tap_hdmi_device_setup(FuDevice *device, GError **error)
+{
+	/* setup device identifier so plugin can disntiguish device during composite_cleaup */
+	fu_device_add_private_flag(device, FU_LOGITECH_TAP_DEVICE_TYPE_HDMI);
+	return fu_logitech_tap_hdmi_device_set_version(device, error);
+}
+
+static gboolean
+fu_logitech_tap_hdmi_device_probe(FuDevice *device, GError **error)
+{
+#if G_USB_CHECK_VERSION(0, 3, 3)
```

Comment:
this check doesn't seem right - what are you using in gusb?

Reply:
Removed